include/pfs/types.hpp

```diff
@@ -454,6 +454,27 @@ struct ip
     ipv6 storage;
 };
 
+struct net_device
```

Comment:
The RX and TX formats are not quite the same.

Both share the first five members: bytes, packets, errs, drop, and fifo along with compressed, which is in different input positions.  frame and multicast are unique to RX while colls and carrier are unique to TX.  

It may be possible to use anonymous unions for pairs:
```
struct net_device
{
    struct stats
    {
        uint64_t bytes;
        uint64_t packets;
        uint64_t errors;
        uint64_t drop;
        uint64_t fifo;
        uint64_t compressed;
        union {
            uint64_t frame;
            uint64_t colls;
        };
        union {
            uint64_t multicast;
            uint64_t carrier;
        };
    };

    std::string name;
    stats rx;
    stats tx;
}
```
However, I do not think anonymous unions are part of C++11, just supported by many compilers. Also, I am not sure how to create a `<<` operator for stats to handle the union names.

Reply:
Thanks! I missed that.
Unions always feel messy, let's avoid them.
If that's the case, let's go with a simple flat struct, just like the one you used to begin with...